src/utilities/createRouter.ts

```diff
@@ -1,41 +1,70 @@
-import { readonly } from 'vue'
+import { DeepReadonly, reactive, readonly } from 'vue'
 import { Resolved, Route, RouteMethods, Routes } from '@/types'
 import { createRouteMethods, createRouterNavigation, resolveRoutes, routeMatch } from '@/utilities'
+import { isBrowser } from '@/utilities/isBrowser'
 import { resolveRoutesRegex } from '@/utilities/resolveRoutesRegex'
 
+type RouterOptions = {
+  initialUrl?: string,
+}
+
 type RouterPush = (url: string, options?: { replace: boolean }) => Promise<void>
 type RouterReplace = (url: string) => Promise<void>
 
 export type Router<
   TRoutes extends Routes
 > = {
   routes: RouteMethods<TRoutes>,
-  route: Readonly<Resolved<Route>>,
+  route: DeepReadonly<Resolved<Route>>,
   push: RouterPush,
   replace: RouterReplace,
   back: () => void,
   forward: () => void,
   go: (delta: number) => void,
 }
 
-export function createRouter<T extends Routes>(routes: T): Router<T> {
+export function createRouter<T extends Routes>(routes: T, options?: RouterOptions): Router<T> {
   const resolved = resolveRoutes(routes)
   const resolvedWithRegex = resolveRoutesRegex(resolved)
   const navigation = createRouterNavigation({
     onLocationUpdate,
   })
 
-  // todo: implement this
-  const route: Router<T>['route'] = readonly({} as any)
+  const route: Resolved<Route> = reactive(getInitialRoute())
 
-  async function onLocationUpdate(url: string): Promise<void> {
-    const match = routeMatch(resolvedWithRegex, url)
+  function getInitialUrl(): string {
+    if (options?.initialUrl) {
+      return options.initialUrl
+    }
+
+    if (isBrowser()) {
+      return window.location.toString()
+    }
```

Comment:
I have a gut feeling that when I see `isBrowser` I want to pull it out into it's own service. I can see that this is maybe a bit extreme here but putting the comment in anyways to get your feedback. The fewer and smaller the tests that require a node and browser version the better IMO. I can supply a PR that makes this change for you to consider

Reply:
https://github.com/kitbagjs/router/pull/45